### PR TITLE
Settings: Cleanup analytics settings from legacy Jetpack version checks

### PR DESCRIPTION
**Pull request by tyxla**
We are supposed to support only Jetpack current version - 1, so we can clean up any older version checks from Calypso. This PR does it for the Google analytics settings.

Part of #29888

#### Changes proposed in this Pull Request

* Cleanup all the Jetpack legacy version checks from Google analytics settings.
* Cleanup any unused props in Google Analytics settings form.

#### Testing instructions

* Checkout this branch, or spin it up on calypso.live.
* Perform the following operations for:
  * A Jetpack site (with Jetpack >= 6.7.0).
  * A WP.com site.
  * An Atomic site.
  * (bonus) a VIP site.
* Go to Traffic settings of the site.
* Verify saving in the Google Analytics card works well just like it did before.
* Compare with production on the same site and verify the same is displayed in the Google Analytics card.

Note: there are some unused selectors that can be deleted after this PR has been merged, but I'll do that in a separate PR (see #29888).

### CODE REVIEW

File: client/my-sites/site-settings/form-analytics.jsx [hunk 1]
```diff
@@ -128,16 +113,11 @@ export class GoogleAnalyticsForm extends Component {
...
 		const showAnonymizeIP =
-			config.isEnabled( 'jetpack/google-analytics-anonymize-ip' ) &&
-			siteIsJetpack &&
-			siteSupportsIPAnonymization;
+			config.isEnabled( 'jetpack/google-analytics-anonymize-ip' ) && siteIsJetpack;
 		const showEnhancedAnalyticsForStores =
-			config.isEnabled( 'jetpack/google-analytics-for-stores-enhanced' ) &&
-			siteIsJetpack &&
-			siteSupportsEnhancedEcommerceTracking;
+			config.isEnabled( 'jetpack/google-analytics-for-stores-enhanced' ) && siteIsJetpack;
```
Comment (jsnajdr) [inline client/my-sites/site-settings/form-analytics.jsx]: Can we remove the `jetpack/google-*` config flags, too? They are enabled in all environments except desktop. Is there a reason to disable them in the desktop Electron app? If I understand correctly, we're just editing and saving Jetpack settings here, not affecting analytics sent by Calypso, right? Cc @allendav who added this code in #17985 and #19888.

Comment (tyxla) [reply client/my-sites/site-settings/form-analytics.jsx]: Good call! I'll take care of that in this PR.

Comment (tyxla) [reply client/my-sites/site-settings/form-analytics.jsx]: The config flags have been completely removed in https://github.com/Automattic/wp-calypso/pull/30042/commits/1b9e675b3b91f082b4d05fbb04c6f48a86fb5a83.

**Comment by tyxla** [comment]
As @jsnajdr suggested (and @allendav confirmed for us), we've removed the `jetpack/google-analytics-*` feature flags completely, as they were no longer necessary.

This PR is ready for another look.

File: client/my-sites/site-settings/form-analytics.jsx [hunk 2]
```diff
@@ -258,11 +217,11 @@ export class GoogleAnalyticsForm extends Component {
 								</FormSettingExplanation>
 							</fieldset>
 						) }
-						{ showAnalyticsForStores && (
+						{ siteIsJetpack && wooCommerceActive && (
 							<FormAnalyticsStores
 								fields={ fields }
 								handleToggleChange={ this.handleToggleChange }
-								showEnhanced={ showEnhancedAnalyticsForStores }
+								showEnhanced={ siteIsJetpack }
```
Comment (jsnajdr) [inline client/my-sites/site-settings/form-analytics.jsx]: nit: `showEnhanced` is effectively always `true`.

File: client/my-sites/site-settings/form-analytics.jsx [hunk 3]
```diff
@@ -308,40 +267,21 @@ export class GoogleAnalyticsForm extends Component {
...
-	);
-	const siteSupportsEnhancedEcommerceTracking = siteSupportsGoogleAnalyticsEnhancedEcommerceTracking(
-		state,
-		siteId
-	);
 	const siteIsJetpack = isJetpackSite( state, siteId );
 	const googleAnalyticsEnabled =
-		site &&
-		( ! siteIsJetpack || ( siteIsJetpack && jetpackModuleActive && jetpackVersionSupportsModule ) );
+		site && ( ! siteIsJetpack || ( siteIsJetpack && jetpackModuleActive ) );
```
Comment (jsnajdr) [inline client/my-sites/site-settings/form-analytics.jsx]: nit: can be compressed into `site && ( ! siteIsJetpack || jetpackModuleActive )`

File: client/my-sites/site-settings/form-analytics.jsx (hunk 2 above)
Comment (tyxla) [reply client/my-sites/site-settings/form-analytics.jsx]: 👍 Cleaned up in https://github.com/Automattic/wp-calypso/pull/30042/commits/cffa1cb93f4a290c044c0f2672b5fd468704e8f2

File: client/my-sites/site-settings/form-analytics.jsx (hunk 3 above)
Comment (tyxla) [reply client/my-sites/site-settings/form-analytics.jsx]: Done in https://github.com/Automattic/wp-calypso/pull/30042/commits/361403632eea4a2cb209e7c46133ad6904a0761c 👍

**Comment by tyxla** [comment]
Thanks for the suggestions @jsnajdr, I've cleaned them up. It's always great to improve as we pass along. 

Would be nice to get a final ✅ before I 🚢 this.

**Comment by jsnajdr** [comment]
> Would be nice to get a final ✅ before I 🚢 this.

Here it is: ✅ :wink:

**Pull request merged**